File: jeehyun/개별문제/SW Expert Academy/1231_중위순회.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

class Solution {
	private static char tree[] = new char[100];
	private static int n;
	
	private static void inorder(int v) {
		if (v <= n) {
			inorder(2 * v);
			System.out.printf("%c", tree[v]);
			inorder(2 * v + 1);
		}
	}
	
	public static void main(String args[]) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int T = 10;
		for (int test_case = 1; test_case <= T; test_case++) {
			n = Integer.parseInt(br.readLine());

			for (int i = 1; i <= n; i++) {
				String info[] = br.readLine().split(" ");
				tree[i] = info[1].charAt(0);
			}

			System.out.print("#" + test_case + " ");
			inorder(1);
			System.out.println();
		}
	}
}

File: jeehyun/개별문제/SW Expert Academy/1231_중위순회.cpp
#include<iostream>
using namespace std;

int n;
char tree[101];

void inorder(int v) {
    if (v > n) return;

    inorder(2 * v);
    cout << tree[v];
    inorder(2 * v + 1);
}

int main(int argc, char** argv)
{
    int T = 10;
    for (int test_case = 1; test_case <= T; ++test_case) {
        cin >> n;

        for (int i = 1; i <= n; i++) {
            cin >> i >> tree[i];

            int left, right;
            if (i < n / 2) {
                cin >> left >> right;
            }
            else if (i == n / 2) {
                cin >> left;
                if (n % 2)
                    cin >> right;
            }
        }

        cout << "#" << test_case << " ";
        inorder(1);
        cout << "\n";
    }
    return 0;
}